tests/Unit/NumericalAlgorithms/Interpolation/Test_ObserveSurfaceIntegrals.cpp

```diff
@@ -0,0 +1,279 @@
+  static std::string name() noexcept { return "TestSolution"; }
+  using type = Scalar<DataVector>;
+};
+struct Square : db::SimpleTag {
+  static std::string name() noexcept { return "Square"; }
+  using type = Scalar<DataVector>;
+};
+struct SquareComputeItem : Square, db::ComputeTag {
+  static std::string name() noexcept { return "Square"; }
+  static Scalar<DataVector> function(const Scalar<DataVector>& x) noexcept {
+    auto result = make_with_value<Scalar<DataVector>>(x, 0.0);
+    get(result) = square(get(x));
+    return result;
+  }
+  using argument_tags = tmpl::list<TestSolution>;
+};
+}  // namespace Tags
+
+template <typename Metavariables, typename InterpolationTargetTag>
+struct MockInterpolationTarget {
+  using metavariables = Metavariables;
+  using chare_type = ActionTesting::MockArrayChare;
+  using array_index = size_t;
+  using action_list = tmpl::list<>;
+  using component_being_mocked =
+      intrp::InterpolationTarget<Metavariables, InterpolationTargetTag>;
+  using initial_databox = db::compute_databox_type<
+      typename intrp::Actions::InitializeInterpolationTarget<
+          InterpolationTargetTag>::template return_tag_list<Metavariables>>;
+
+  using const_global_cache_tag_list = Parallel::get_const_global_cache_tags<
+      tmpl::list<typename InterpolationTargetTag::compute_target_points,
+                 typename InterpolationTargetTag::post_interpolation_callback>>;
+};
+
+template <typename Metavariables>
+struct MockInterpolator {
+  using metavariables = Metavariables;
+  using chare_type = ActionTesting::MockArrayChare;
+  using array_index = size_t;
+  using const_global_cache_tag_list = tmpl::list<>;
+  using action_list = tmpl::list<>;
+  using component_being_mocked = intrp::Interpolator<Metavariables>;
+  using initial_databox =
+      db::compute_databox_type<typename intrp::Actions::InitializeInterpolator::
+                                   template return_tag_list<Metavariables>>;
+};
+
+struct MockMetavariables {
+  struct InterpolationTargetA {
+    using compute_items_on_source = tmpl::list<>;
+    using vars_to_interpolate_to_target =
+        tmpl::list<Tags::TestSolution,
+                   gr::Tags::SpatialMetric<3, Frame::Inertial>>;
+    using compute_items_on_target = tmpl::list<Tags::SquareComputeItem>;
+    using compute_target_points =
+        intrp::Actions::KerrHorizon<InterpolationTargetA, ::Frame::Inertial>;
+    using post_interpolation_callback =
+      intrp::callbacks::ObserveSurfaceIntegrals<
+            tmpl::list<Tags::Square>, observers::OptionTags::ReductionFileName,
+            Frame::Inertial>;
+    // This `type` is so this tag can be used to read options.
+    using type = typename compute_target_points::options_type;
+  };
+  using interpolator_source_vars =
+      tmpl::list<Tags::TestSolution,
+                 gr::Tags::SpatialMetric<3, Frame::Inertial>>;
+  using interpolation_target_tags = tmpl::list<InterpolationTargetA>;
+  using temporal_id = Time;
```

Comment:
Evolutions don't use `Time` here, they use `TimeId`.  Is this a different metavariables than for the evolution, or do they have to be made compatible?

Reply:
I was thinking this would be the same metavariables as for the evolution (I didn't know you can have two different metavariables in the same executable).  Nothing in the interpolation structure should care whether temporal_id is Time, TimeId, or something else (as long as it can be a key in a map, is puppable, and has equality/inequality operators).

Reply:
I switched to TimeId.